changelog/next/changes/4349--subscribe-metrics.md

```diff
@@ -0,0 +1,2 @@
+Pipeline activity for pipelines starting with `subscribe | where <expr>` will no
+longer report ingress that does not match the provided filter expression.
```

Comment:
Yds. I talked with @tobim about this previously, because in prod deployments when you use SPMC with different filter expressions as a poor mans switch-case statement the metrics end up being rather confusing without this change.

Also, in the future subscribe will be backed by some form of catalog, and that will have also optimize the filter expression, leading to the same result.

Reply:
Hmm... I'm not sure whether I really like this, but let's proceed like that for now.

Reply:
It also makes `export --live` and `subscribe` behave the same way, maybe that convinces you? 😅

Reply:
Maybe they are wrong too? 🤓 Nah, it's fine. There are pros and cons to both approaches.